packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -277,7 +276,6 @@ const devHtmlHook: IndexHtmlTransformHook = async (
 
       // ensure module in graph after successful load
       const mod = await moduleGraph.ensureEntryFromUrl(url, false)
-      ensureWatchedFile(watcher, mod.file, config.root)
```

Comment:
Just wondering, how is this covered in Rollup 4? Or is it related to the way we call `this.addWatchFile` explicitly now? 

Seems like this is also done in `transformRequest.ts`.

Reply:
Rollup 4 only adds files that was loaded by the default `load` hook (in other words, no plugins handled that `id` with the `load` hook) (https://github.com/rollup/rollup/pull/5150). Rollup 3 added files even if it was loaded by a plugin.

Vite's plugin container doesn't have a default `load` hook but instead it has `vite:load-fallback` plugin. This plugin works like the default `load` hook and `this.addWatchFile` is now called there. This part covers the `transformRequest.ts` one.

That said, I noticed that this `ensureWatchedFile` in `indexHtml.ts` needs to be there as it doesn't go through the whole plugin pipeline. Thanks for catching! I'll restore this one.

Reply:
Thanks for the explanation!